python/pyspark/sql/tests.py

```diff
@@ -1964,6 +1964,16 @@ def test_replace(self):
                .replace(False, True).first())
         self.assertTupleEqual(row, (True, True))
 
+        # replace with None
+        row = self.spark.createDataFrame(
+            [(u'Alice', 10, 80.0)], schema).replace(u'Alice', None).first()
+        self.assertTupleEqual(row, (None, 10, 80.0))
+
+        # replace with numerics and None
+        row = self.spark.createDataFrame(
+            [(u'Alice', 10, 80.0)], schema).replace([10, 80], [20, None]).first()
+        self.assertTupleEqual(row, (u'Alice', 20, None))
```

Comment:
Can you add a test that `to_replace` is a list and `value` is not given (so as default value `None`)? Previously this will arise a `ValueError`. But now it is a valid usage. We are better to add a test explicitly for it.